docs/client_reference.rst

```diff
@@ -0,0 +1,261 @@
+                   ``None`` by default (optional).
+
+   :param bool expect100: Expect 100-continue response from server.
+                          ``False`` by default (optional).
+
+   :param aiohttp.connector.BaseConnector connector: BaseConnector sub-class
+                                                     instance to support connection pooling.
+
+   :param bool read_until_eof: Read response until eof if response
+                               does not have Content-Length header.
+                               ``True`` by default (optional).
+
+   :param request_class: Custom Request class implementation (optional)
+
+   :param response_class: Custom Response class implementation (optional)
+
+   :param loop: :ref:`event loop<asyncio-event-loop>`
+                used for processing HTTP requests.
+                If param is ``None``, :func:`asyncio.get_event_loop`
+                is used for getting default event loop, but we strongly
+                recommend to use explicit loops everywhere.
+                (optional)
+
+
+Usage::
+
+     >>> import aiohttp
+     >>> resp = yield from aiohttp.request('GET', 'http://python.org/')
+     >>> resp
+     <ClientResponse(python.org/) [200]>
+     >>> data = yield from resp.read()
+
+
+Client Session
+--------------
+
+.. py:class:: ClientSession(*, connector=None, loop=None, request_class=None, response_class=None, cookies=None, headers=None, auth=None)
+   :module: aiohttp.client
+
+   The class for creating client sessions and making requests.
+
+   :param aiohttp.connector.BaseConnector connector: BaseConnector sub-class
+                                                     instance to support connection pooling.
+
+
+   :param loop: :ref:`event loop<asyncio-event-loop>`
+            used for processing HTTP requests.
+            If param is ``None``, :func:`asyncio.get_event_loop`
+            is used for getting default event loop, but we strongly
+            recommend to use explicit loops everywhere.
+            (optional)
+
+
+   :param request_class: Custom Request class implementation (optional)
+
+   :param response_class: Custom Response class implementation (optional)
+
+   :param dict cookies: Cookies to send with the request (optional)
+
+   :param dict headers: Dictionary of HTTP Headers to send with
+                        the request (optional)
+
+   :param aiohttp.helpers.BasicAuth auth: BasicAuth named tuple that represents
+                                          HTTP Basic Auth (optional)
+
+
+   .. py:method:: ClientSession.request(method, url, *, params=None, data=None, headers=None, auth=None, allow_redirects=True, max_redirects=10, encoding='utf-8', version=HttpVersion(major=1, minor=1), compress=None, chunked=None, expect100=False, read_until_eof=True)
+      :module: aiohttp.client
+
+      Performs an asyncronous http request. Returns a response object.
+
+      The function is a :ref:`coroutine <coroutine>`.
```

Comment:
Also, latest sphinx it seems knows how to highlight coroutine functions. Or is it just doc.python.org improvement?